raiden/network/rpc/client.py

```diff
@@ -621,6 +621,9 @@ def new_netting_channel(self, peer1, peer2, settle_timeout):
         else:
             other = peer1
 
+        if self.proxy.getChannelWith(other) != '0' * 40:
```

Comment:
I don't think this is the best solution. I can see at least one case where this would not work even though it should. This is due to the "not optimal" way that we are currently deleting a channel. This check would fail after a channel has been settled and two parties then try to open a new channel. My suggestion would be to make the check in the `self.proxy.newChannel.transact` line below, since that line will throw if a channel already exists.